### PR TITLE
Fixes instantiation of CertificateAuthentication object from trino-python-client

### PR DESCRIPTION
**Pull request by KateShopify**
## Overview

resolves https://github.com/starburstdata/dbt-trino/issues/149 To work with certificate authentication in [`trino-python-client`](https://github.com/trinodb/trino-python-client)
Not updating any functionality or function signatures, so the README should not need an update. 
## Checklist

- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] `README.md` updated and added information about my change
- [x] I have run `changie new` to [create a changelog entry](https://github.com/starburstdata/dbt-trino/blob/master/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

File: .changes/unreleased/Fixes-20221009-132438.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+kind: Fixes
+time: 2022-10-09T13:24:38.264985+02:00
```
Comment (findinpath) [inline .changes/unreleased/Fixes-20221009-132438.yaml:2]: Please do add a `body` summarizing your changes.

**Comment by findinpath** [comment]
`removing named arguments to match function signature` -> `Remove named arguments to match function signature`

**Comment by findinpath** [comment]
Please squash your changes.

File: dbt/adapters/trino/connections.py [hunk 2]
```diff
@@ -131,8 +131,8 @@ def method(self):
 
     def trino_auth(self):
         return trino.auth.CertificateAuthentication(
```
Comment (findinpath) [inline dbt/adapters/trino/connections.py:133]: Let's add a unit test to verify the output of `trino_auth()` in
  `tests.unit.test_adapter.TestTrinoAdapterAuthenticationMethods.test_certificate_authentication`

**Comment by mdesmet** [comment]
@KateShopify : Please execute `pre-commit install` to install the pre-commit hooks. That ensures the formatting & linting run on commit and saves you from failing the build for those reasons. :)

File: .changes/unreleased/Fixes-20221009-132438.yaml (hunk 1 above)
Comment (KateShopify) [reply .changes/unreleased/Fixes-20221009-132438.yaml:3]: updated 👍

**Review by hovaesco — APPROVED** [review]
Hi @KateShopify, please change commit message to `Fixes instantiation of CertificateAuthentication object from trino-python-client`, otherwise looks good.

**Comment by findinpath** [comment]
```
adding changie log

updating test
```

This part of the commit log is not really relevant. Please remove it.

File: tests/unit/test_adapter.py [hunk 3]
```diff
@@ -265,7 +265,9 @@ def test_certificate_authentication(self):
             }
         )
         credentials = connection.credentials
+        trino_auth = credentials.trino_auth()
         self.assertIsInstance(credentials, TrinoCertificateCredentials)
+        self.assertIsInstance(trino_auth, trino.auth.CertificateAuthentication)
```
Comment (findinpath) [inline tests/unit/test_adapter.py:270]: We can as well assert equality.

`self.assertEqual(credentials.trino_auth(), ....)`

File: tests/unit/test_adapter.py [hunk 4]
```diff
@@ -265,7 +265,9 @@ def test_certificate_authentication(self):
             }
         )
         credentials = connection.credentials
+        trino_auth = credentials.trino_auth()
```
Comment (findinpath) [inline tests/unit/test_adapter.py:268]: No need to create a new variable. Inline the operation directly within the assertion

File: tests/unit/test_adapter.py (hunk 3 above)
Comment (KateShopify) [reply tests/unit/test_adapter.py]: added 👍

File: tests/unit/test_adapter.py (hunk 4 above)
Comment (KateShopify) [reply tests/unit/test_adapter.py]: updated 👍

File: tests/unit/test_adapter.py [hunk 5]
```diff
@@ -266,6 +266,11 @@ def test_certificate_authentication(self):
         )
         credentials = connection.credentials
         self.assertIsInstance(credentials, TrinoCertificateCredentials)
+        self.assertIsInstance(credentials.trino_auth(), trino.auth.CertificateAuthentication)
+        self.assertEqual(
+            credentials.trino_auth(),
+            trino.auth.CertificateAuthentication("/path/to/client_cert", "password"),
```
Comment (findinpath) [inline tests/unit/test_adapter.py:272]: What do we need for the last `,` ?

**Review by findinpath — APPROVED** [review]

**Comment by hovaesco** [comment]
Thank you for your contribution @KateShopify

**Pull request merged**